pkg/build/controller/factory/factory.go

```diff
@@ -263,6 +332,59 @@ func (lw *buildLW) Watch(resourceVersion string) (watch.Interface, error) {
 	return lw.client.Builds(kapi.NamespaceAll).Watch(labels.Everything(), fields.Everything(), resourceVersion)
 }
 
+// buildDeleteLW is a ListWatcher implementation that watches for builds being deleted
+type buildDeleteLW struct {
+	client ControllerClient
+	store  cache.Store
+}
+
+// List lists all Builds.
+func (lw *buildDeleteLW) List() (runtime.Object, error) {
+	glog.V(5).Infof("Checking for deleted builds")
+	selector := labels.LabelSelector{}.Add(buildapi.IsBuildPodLabel, labels.EqualsOperator, []string{"true"})
+	podList, err := lw.client.KubeClient.Pods(kapi.NamespaceAll).List(selector, fields.Everything())
```

Comment:
Any reason you can't list pods AND builds and then only send the builds that were deleted into the controller?